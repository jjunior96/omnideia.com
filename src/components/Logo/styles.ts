import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import { LogoProps } from '.';

const containerModifiers = {
  normal: () => css`
    width: 11rem;
    height: 3.3rem;
  `,

  large: () => css`
    width: 20rem;
    height: 5.9rem;
  `,

  hideOnMobile: () => css`
    ${media.lessThan('medium')`
      width: 5.8rem;
      height: 4.5rem;

      img {
        height: 4.5rem;
        pointer-events: none;
      }
      .text {
        display: none;
      }
    `}
  `
};

export const Container = styled.div<LogoProps>`
  ${({ theme, color, size, hideOnMobile }) => css`
    color: ${theme.colors[color!]};

    svg {
      color: ${theme.colors.purple_300};

      width: 3.2rem;
      height: 3.2rem;
    }

    ${!!size && containerModifiers[size]}
    ${!!hideOnMobile && containerModifiers.hideOnMobile}
  `}
`;
