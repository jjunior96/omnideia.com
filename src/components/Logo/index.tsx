import Link from 'next/link';
import { BsLightbulb as IdeaIcon } from 'react-icons/bs';

import * as S from './styles';

export type LogoProps = {
  color?: 'white' | 'black';
  size?: 'normal' | 'large';
  hideOnMobile?: boolean;
  id?: string;
};

const Logo = ({
  color = 'white',
  size = 'normal',
  hideOnMobile = false
}: LogoProps) => (
  <S.Container color={color} size={size} hideOnMobile={hideOnMobile}>
    <Link href="/">
      <a>
        <IdeaIcon />
      </a>
    </Link>
  </S.Container>
);

export default Logo;
