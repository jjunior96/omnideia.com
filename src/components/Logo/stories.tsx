import { Story, Meta } from '@storybook/react/types-6-0';

import Logo, { LogoProps } from '.';

export default {
  title: 'Logo',
  component: Logo,
  parameters: {
    backgrounds: {
      default: 'white'
    }
  }
} as Meta;

export const Default: Story<LogoProps> = (args) => (
  <div
    style={{
      width: '40rem',
      height: '25rem',
      margin: '0 auto',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    }}
  >
    <Logo {...args} />
  </div>
);

Default.args = {
  color: 'black',
  size: 'normal'
};
