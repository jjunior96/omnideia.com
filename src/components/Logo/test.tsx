import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Logo from '.';

describe('<Logo />', () => {
  it('should render a white label by default', () => {
    const { container } = renderWithTheme(<Logo />);

    // 1 - Pega o elemento que tem um LabelText = Won Games
    // 2 - Com o `parantElement`, pega o pai do elemento com LabelText Won Games
    // 3 - Verifica se o elemento pai tem o estilo esperado
    expect(screen.getByLabelText(/Won Games/i).parentElement).toHaveStyle({
      color: '#FAFAFA'
    });

    expect(container).toMatchSnapshot();
  });

  it('should render a black label when color is passed', () => {
    renderWithTheme(<Logo color="black" />);

    expect(screen.getByLabelText(/Won Games/i).parentElement).toHaveStyle({
      color: '#0D0D0D'
    });
  });

  it('should render with size default', () => {
    renderWithTheme(<Logo />);

    expect(screen.getByLabelText(/Won Games/i).parentElement).toHaveStyle({
      width: '11rem'
    });
  });

  it('should render with size large', () => {
    renderWithTheme(<Logo size="large" />);

    expect(screen.getByLabelText(/Won Games/i).parentElement).toHaveStyle({
      width: '20rem'
    });
  });
});
