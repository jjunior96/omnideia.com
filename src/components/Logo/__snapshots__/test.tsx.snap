// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render a white label by default 1`] = `
.c0 {
  color: #FAFAFA;
  position: relative;
  width: 11rem;
  height: 3.3rem;
}

.c0 svg {
  color: #5855E9;
  width: 3.2rem;
  height: 3.2rem;
}

.c1 {
  position: absolute;
  pointer-events: none;
  width: 1.4rem;
  height: 1.4rem;
  border-radius: 8px;
  left: 1rem;
  top: 0.3rem;
}

<div>
  <div
    class="c0"
    color="white"
  >
    <div
      class="c1"
    />
    <a
      href="/"
    >
      <svg
        data-testid="Logo"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 6a6 6 0 1 1 10.174 4.31c-.203.196-.359.4-.453.619l-.762 1.769A.5.5 0 0 1 10.5 13a.5.5 0 0 1 0 1 .5.5 0 0 1 0 1l-.224.447a1 1 0 0 1-.894.553H6.618a1 1 0 0 1-.894-.553L5.5 15a.5.5 0 0 1 0-1 .5.5 0 0 1 0-1 .5.5 0 0 1-.46-.302l-.761-1.77a1.964 1.964 0 0 0-.453-.618A5.984 5.984 0 0 1 2 6zm6-5a5 5 0 0 0-3.479 8.592c.263.254.514.564.676.941L5.83 12h4.342l.632-1.467c.162-.377.413-.687.676-.941A5 5 0 0 0 8 1z"
        />
      </svg>
    </a>
  </div>
</div>
`;
